viewing_party/party.py

```diff
@@ -1,23 +1,171 @@
+        for movie in range (len(user_data["friends"][i]["watched"])): # Loops through each friend's nested movie list
+            friend_movie = user_data["friends"][i]["watched"][movie]
+            if friend_movie in user_data["watched"]:
+                user_same_list.append(friend_movie)
+
+    for i in range(len(user_data["watched"])):
+        if user_data["watched"][i] not in user_same_list:
+            unique_list.append(user_data["watched"][i])
+
+    return unique_list
+
+def get_friends_unique_watched(user_data):
+
+    friends_unique_list = []
+
+    for i in range (len(user_data["friends"])): 
+        for movie in range (len(user_data["friends"][i]["watched"])):
+            friend_movie = user_data["friends"][i]["watched"][movie]
+            if friend_movie not in user_data["watched"] and friend_movie not in friends_unique_list:
+                    friends_unique_list.append(friend_movie)
+
+    return friends_unique_list
+
         
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+
+def get_available_recs(user_data):
+
+    recommended_list = []
+
+    for i in range (len(user_data["friends"])): 
+        for movie in range (len(user_data["friends"][i]["watched"])): 
+                friend_movie = user_data["friends"][i]["watched"][movie]
+                friend_host = friend_movie["host"]
+                friends_unique_list = get_friends_unique_watched(user_data)
+
+                if friend_host in user_data["subscriptions"] and friend_movie in friends_unique_list:
+                    recommended_list.append(user_data["friends"][i]["watched"][movie])
+    
+    return recommended_list
+                    
+
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(user_data):
+    
+    recommended_list = []
+
+    if len(user_data["watched"])>0:
+        user_genre = user_data["watched"][0]["genre"]
+
+    else:
+        user_genre = None 
+
+    for i in range (len(user_data["friends"])): 
+        for movie in range (len(user_data["friends"][i]["watched"])): 
+            friend_movie = user_data["friends"][i]["watched"][movie]
+            friend_genre = friend_movie["genre"]
+            friends_unique_list = get_friends_unique_watched(user_data)
+    
+            if friend_genre == user_genre and friend_movie in friends_unique_list:
+                recommended_list.append(user_data["friends"][i]["watched"][movie])
+            
+    return recommended_list
+
+def get_rec_from_favorites(user_data):
+
+    fav_recs =[]
+
+    for i in range (len(user_data["favorites"])):
+        user_unique_list = get_unique_watched(user_data)
+        movie = user_data["favorites"][i]
+        if movie in user_unique_list:
+            fav_recs.append(movie)
+
+    return fav_recs
```

Comment:
👍  Good work user helper functions!! Here are a few things to consider:

- How would this function look using a `for..in` loop? 
- Do we need to call `get_unique_watched` in every iteration?